### PR TITLE
UCT/IB: Fix compilation without dv

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/configure.m4 [hunk 1]
```diff
@@ -383,13 +380,14 @@ AS_IF([test "x$with_ib" = "xyes"],
...
-       AS_IF([test "x$with_ib_hw_tm" != xno],
-           [AC_CHECK_HEADERS([infiniband/tm_types.h])
+       AS_IF([test "x$with_ib_hw_tm" != xno], [
             AC_CHECK_MEMBER([struct ibv_exp_tmh.tag], [with_ib_hw_tm=exp], [],
                             [[#include <infiniband/verbs_exp.h>]])
-            AC_CHECK_MEMBER([struct ibv_tmh.tag], [with_ib_hw_tm=upstream], [],
-                            [[#include <infiniband/tm_types.h>]])
-           ])
+            AS_IF([test "x$with_mlx5_dv" = "xyes"], [
+                AC_CHECK_HEADERS([infiniband/tm_types.h])
```
Comment (yosefe) [inline src/uct/ib/configure.m4:387]: can remove?

File: src/uct/ib/rdmacm/rdmacm_cm.h [hunk 2]
```diff
@@ -10,8 +10,11 @@
 #include <uct/base/uct_cm.h>
 #include <ucs/datastruct/khash.h>
 #include <ucs/sys/string.h>
+#include <ucs/type/spinlock.h>
 #include <ucs/datastruct/bitmap.h>
+#if HAVE_MLX5_HW
```
Comment (yosefe) [inline src/uct/ib/rdmacm/rdmacm_cm.h:15]: double check if need `#if` around `struct mlx5dv_devx_obj` below

File: src/uct/ib/configure.m4 (hunk 1 above)
Comment (Artemy-Mellanox) [reply src/uct/ib/configure.m4:387]: it adds HAVE_INFINIBAND_TM_TYPES_H
later we use it create upstream/exp compatible headers
https://github.com/openucx/ucx/blob/master/src/uct/ib/rc/accel/rc_mlx5_common.h#L26

**Review by yosefe — APPROVED** [review]
pls squash without rebase

**Pull request merged**